pages/docs/guides/message-validation.md

```diff
@@ -0,0 +1,125 @@
+- Consumer: The consumer is responsible for getting the producer's messages.
+- Broker: The broker acts as a bridge between the consumer and the producer because messages travel through the broker.
+
+Message validation can occur in different places in your system. This guide highlights three of those:
+- Both producers and consumers can do validation internally in runtime.
+- Validation of the message can be handled by API Gateway
+- Validation of messages can be a native solution implemented by the schema registry.
+Because consumers and producers cannot communicate directly, the AsyncAPI file dictates what should be included in the payload when a service produces a message. The AsyncAPI document also tells the consumer about the message's properties.
+Let's further break down how validation works for all.
+
+### Validation in runtime
+Messages produced and consumed are both required for runtime message validation. The AsyncAPI document should include descriptions of payload schemas so that you can read them in your application and validate messages that are consumed and produced by the application.
+
+Before messages reach the consumer, runtime validation ensures that any errors are resolved and valid messages are sent to your application.
+
+An example implementation of message validation in runtime is [asyncapi-validator](https://github.com/WaleedAshraf/asyncapi-validator) library that enables you to validate messages produced/consumed in your application against schemas provided in your AsyncAPI document.
+Check out the Message validation in the runtime() tutorial if you want to get your hands dirty with message validation. 
+
+```mermaid
+graph TD
+    subgraph Producer
+    G[Message produced] -->|Message| B{Is Message valid?}
+    B -->|No| A[Logs]
+    end
+    B -->|Yes| C[Broker]
+    C -->|Message| E{Is Message valid?}
+    subgraph Consumer
+    E -->|No| F[Logs]
+    E -->|Yes| H[Message consumed]
+    end
+```
+
+### Validation in a gateway
+A gateway intercepts all incoming messages and routes them through the middleware and handler pipelines. The gateway sits between the producer and the broker. The messages are routed through the gateway first, and the gateway determines whether the message is valid. If the message is invalid, it displays an error and is not forwarded to the broker.
+
+An example implementation of message validation in a gateway is the [AsyncAPI gateway](https://github.com/asyncapi/event-gateway). It intercepts all incoming messages moving them into a pipeline of middlewares and handlers such as message validation. You can use a Kafka consumer/producer[(kcat)](https://github.com/edenhill/kcat), a broker, and a simple WebSocket to run the AsyncAPI gateway in your machine.
+Check out an [AsyncAPI file demo with Studio.](https://studio.asyncapi.com/?url=https://raw.githubusercontent.com/asyncapi/event-gateway/master/deployments/k8s/event-gateway-demo/event-gateway-demo.asyncapi.yaml)
```

Comment:
"Check out an AsyncAPI file demo with Studio."

Why? How is this related to the Gateway validation topic? Or the previously mentioned Kafka protocol?

Let's expand this sentence and explain why this demo file is relevant in this section.